### PR TITLE
[bitnami/postgresql-ha] set some bits compatible with pss restricted

### PR DESCRIPTION
**Pull request by jcpunk**
### Description of the change

These enhanced security defaults bring the chart closer into line with the security best practices.

### Benefits

Better pod security by default following the kubernetes best practices.

### Possible drawbacks

Folks doing local manipulations of the container may encounter surprises.

### Applicable issues

- fixes #

### Additional information

Because the init container requires root, this cannot be fully run within a restricted namespace.

### Checklist

- [x] Chart version bumped in `Chart.yaml` according to [semver](http://semver.org/). This is *not necessary* when the changes only affect README.md files.
- [x] Variables are documented in the values.yaml and added to the `README.md` using [readme-generator-for-helm](https://github.com/bitnami-labs/readme-generator-for-helm)
- [x] Title of the pull request follows this pattern [bitnami/<name_of_the_chart>] Descriptive title
- [x] All commits signed off and in agreement of [Developer Certificate of Origin (DCO)](https://github.com/bitnami/charts/blob/main/CONTRIBUTING.md#sign-your-work)

### CODE REVIEW

**Comment by carrodher** [comment]
Thanks for your contribution, it seems a new version was released in the meantime. Could you please solve the conflicts? Thanks!

**Comment by jcpunk** [comment]
rebased

**Comment by migruiz4** [comment]
Adding @andresbono to make sure these PRs are all aligned:

- https://github.com/bitnami/charts/pull/17387
- https://github.com/bitnami/charts/pull/17388
- https://github.com/bitnami/charts/pull/17291
- https://github.com/bitnami/charts/pull/17291

Although, we have yet to evaluate associated risks, such as the impact of these changes on backward compatibility with older versions of Kubernetes.

File: bitnami/postgresql-ha/values.yaml [hunk 1]
```diff
@@ -1550,10 +1583,17 @@ metrics:
...
+  ## @param metrics.podSecurityContext.runAsGroup Group ID for the PostgreSQL Prometheus exporter container
+  ## @param metrics.podSecurityContext.runAsNonRoot Set PostgreSQL Prometheus exporter container's Security Context runAsNonRoot
+  ## @param metrics.podSecurityContext.seccompProfile.type Set PostgreSQL Prometheus exporter  container's Security Context seccompProfile
   ##
   podSecurityContext:
     enabled: true
     runAsUser: 1001
+    runAsGroup: 0
+    runAsNonRoot: true
+    seccompProfile:
```
Comment (andresbono) [inline bitnami/postgresql-ha/values.yaml:1595]: Should we include the following container security context constraints here? Is there any reason to not include them for this container?

```
    allowPrivilegeEscalation: false
    capabilities:
      drop:
        - ALL
```

Comment (jcpunk) [reply bitnami/postgresql-ha/values.yaml:1595]: I believe that was an oversight on my end.  I'll get those added and the branch rebased.

Comment (andresbono) [reply bitnami/postgresql-ha/values.yaml:1595]: You are right, I did not notice this one was for the PodSecurityContext. 👍

**Review by andresbono — DISMISSED** [review]

**Review by andresbono — APPROVED** [review]

**Pull request merged**